lib/RestructureCFG/InlineDivergentScopesPass.cpp

```diff
@@ -305,6 +305,14 @@ static void performIDS(DivergenceDescriptor &DivergenceDescriptor,
   IRBuilder<> TailBuilder(Tail);
   TailBuilder.Insert(TailTerminator);
 
+  // Populate the `DivergentSuccessors` summing all ones present in each
+  // `DivergenceDescriptor`
+  SmallSet<BasicBlock *, 4> DivergentSuccessors;
+  for (auto Divergence : MultipleDivergences) {
+    DivergentSuccessors.insert(Divergence.DivergentSuccessors.begin(),
+                               Divergence.DivergentSuccessors.end());
+  }
```

Comment:
Doesn't copy-assigning work?

Reply:
Mmmm I'm not sure how you want to use the copy-assignment here. In `DivergentSuccessors` I need to put all the elements inside the `DivergentSuccessors` field of each `Divergence` I iterate on.

Using the copy-assignment would be possible only once, what am I missing?

Reply:
Nevermind.